libraries/AP_NavEKF3/AP_NavEKF3_MagFusion.cpp

```diff
@@ -206,6 +203,24 @@ void NavEKF3_core::SelectMagFusion()
         yaw_source_reset = true;
     }
 
+    // Store yaw angle when moving for use as a static reference when not moving
+    if (!onGroundNotMoving) {
+            if (fabsf(prevTnb[0][2]) < fabsf(prevTnb[1][2])) {
+                // A 321 rotation order is best conditioned because the X axis is closer to horizontal than the Y axis
+                yawAngDataStatic.order = rotationOrder::TAIT_BRYAN_321;
+            } else {
+                // A 312 rotation order is best conditioned because the Y axis is closer to horizontal than the X axis
+                yawAngDataStatic.order = rotationOrder::TAIT_BRYAN_312;
+            }
+            if (yawAngDataStatic.order == rotationOrder::TAIT_BRYAN_321) {
```

Comment:
There's a small indenting problem in this section.  Also we can simplify the logic a bit.  I don't think we need two if/else statements after each other.  just one should do I think.  I've added a commit in my branch that makes these two changes.

Reply:
I've pulled in your fix. TXS.